DateTimeCalendar/Calendar.java

```diff
@@ -0,0 +1,135 @@
+public class Calendar{
+    /*
+    Goals:
+    - When created, indicate date and current day of the week (Monday, Tuesday, Wednesday, etc…)
+    - Given another date tell which day of the week it was or it will be.
+    - Calculate number of weeks within a month of a certain year. A week counts if all seven 
+    days (monday to sunday) happen within the same month; if the month starts in the middle of 
+    the week that week is not considered for the number returned by this method.
+    */ 
+
+    public enum DayOfWeek {
+        SUN("Sunday"),
+        MON("Monday"),
+        TUE("Tuesday"),
+        WED("Wednesday"),
+        THR("Thursday"),
+        FRI("Friday"),
+        SAT("Saturday"),
+        ;
+
+        public final String longName;
+
+        private DayOfWeek(final String longName){
+            this.longName = longName;
+        }
+    }
+
+    private int currentDayOfWeek;
+    private Date date;
+
+    private Calendar(){
+
+    }
+
+    public Calendar(Date date){
+        setDate(date);
+        setCurrentDayOfTheWeek(date);
+    }
+
+    public static int calcWeeksInMonth(Date date){
+        //a week is considered mon-sun
+        //Example: 3/1/1900
+
+        final Date firstOfTheMonth = new Date(date.getYear(), date.getMonth(), 1); // creating obj to represent first of the month
+        final int firstWeekDayOfMonth = calcDayOfWeek(firstOfTheMonth); // calculate first week day of the month -> 4 (Thursday)
+        int monthDayCount = date.getMonth().days; // 31
+
+        //check for month feb and for leap year
+        if (date.getMonth().ordinal() == 1 && Date.isLeapYear(date.getYear())) {
+            monthDayCount++;
+        }
+
+        // int firstWeekDayOfMonth = calcDayOfWeek(firstOfTheMonth); // calculate first week day of the month -> 4 (Thursday)
+
+        //want to use first day of the month as a place holder while we tick ahead to tell us the day of the first monday
+        int firstMondayOfMonth = 1; //Every month starts on day 1
+        int dayCountToFirstMonday = firstWeekDayOfMonth; //want to preserve firstWeekDayOfMonth but need to have a value to incremtnt.
```

Comment:
You don't need this variable anymore, do you?

Reply:
I'm not sure I follow what your suggesting. Can you please elaborate?